### DISCUSSION TITLE
Way to get custom ID of a button?

### DISCUSSION THREAD

**Discussion by Robin5605** [Rapptz/discord.py#7082] [Q&A] [answered]
First time using buttons, so sorry if the question is stupid. Anyway, I have multiple buttons in a message and want to know if there's a way to differentiate them. I have 3 buttons, and when a user clicks one, I'd like a message that says "user has clicked x button." Specifically, I'm wondering if there's a way to get the custom ID of a button, it's label, or some other property that distinguishes which button was clicked.

**Comment by imvaskel** [comment, marked as answer]
Buttons feature a `callback` function that you can define in the class to handle callbacks from buttons. 
Buttons also have the `custom_id` attribute.
Docs [here](https://discordpy.readthedocs.io/en/master/api.html#discord.ui.Button.callback).

**Reply by Robin5605** [reply]
Great, thanks for the reply! Would you mind giving me a small snippet? I'm still confused on it. Thanks!

**Reply by imvaskel** [reply]
[Here](https://gist.github.com/Rapptz/dbfd8cd945a9245e5504a54c2b9eda03) is a tictactoe program made by Rapptz with buttons.

**Reply by Robin5605** [reply]
Awesome! Will check it out. Thanks again!
